libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -444,6 +451,10 @@ void AP_Baro::init(void)
         _user_ground_temperature.notify();
     }
 
+    if (!_atm) {
+        _atm = new UR_Atmosphere;
```

Comment:
do we really need the singleton and pointer ? isn't just a class member enought ? as it won't be use outside of AP_Baro

Reply:
This is a circular initialization, this ensure the atm initialize if baro isn't yet, and viceversa. Atm is focused to be used mainly.